File: mobile/lib/features/sms_detection/domain/entities/sms_pending_entity.dart
import 'package:equatable/equatable.dart';

class SmsPendingEntity extends Equatable {
  final String id;
  final String userId;
  final String smsContent;
  final String? extractedMerchant;
  final double? extractedAmount;
  final DateTime? extractedDate;
  final String? extractedAccount;
  final String status; // PENDING, ACCEPTED, REJECTED, IGNORED
  final DateTime createdAt;
  final DateTime updatedAt;

  const SmsPendingEntity({
    required this.id,
    required this.userId,
    required this.smsContent,
    this.extractedMerchant,
    this.extractedAmount,
    this.extractedDate,
    this.extractedAccount,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
  });

  @override
  List<Object?> get props => [
        id,
        userId,
        smsContent,
        extractedMerchant,
        extractedAmount,
        extractedDate,
        extractedAccount,
        status,
        createdAt,
        updatedAt,
      ];
}

File: backend/src/modules/sms-review/entities/sms-pending.entity.ts
export class SmsPendingEntity {
  id!: string;
  userId!: string;
  smsContent!: string;
  extractedMerchant?: string | null;
  extractedAmount?: number | null;
  extractedDate?: Date | null;
  extractedAccount?: string | null;
  status!: string; // PENDING, ACCEPTED, REJECTED, IGNORED
  createdAt!: Date;
  updatedAt!: Date;

  constructor(partial: Partial<SmsPendingEntity>) {
    Object.assign(this, partial);
  }
}
